sdk/src/types/block/core/validation.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use packable::{
+    error::{UnpackError, UnpackErrorExt},
+    packer::Packer,
+    unpacker::Unpacker,
+    Packable,
+};
+
+use crate::types::block::{
+    core::{verify_parents, Block},
+    parent::{ShallowLikeParents, StrongParents, WeakParents},
+    protocol::{ProtocolParameters, ProtocolParametersHash},
+    Error,
+};
+
+/// A builder for a [`ValidationBlock`].
+pub struct ValidationBlockBuilder {
+    strong_parents: StrongParents,
```

Comment:
While we're at it, we should distinguish these parent types from basic block, which have different bounds.

Reply:
Put type aliases in the respective modules to allow for specific bounds